components/header.js

```diff
@@ -1,67 +1,189 @@
 
-    header {
-      width: 80%;
-      margin: 0 auto;
-      justify-content: space-between;
-      display: flex;
-      align-items: center;
-    }
+class Header extends HTMLElement {
+  constructor() {
+    super();
+    this.render();
+    this.setupEventListeners();
+    this.handleResize(); // Ensure the menu is closed initially if needed
+  }
 
-    .buttons-container {
-      display: flex;
-      align-items: center;
-      column-gap: 40px;
-    }
+  setupEventListeners() {
+    const menuButton = this.shadowRoot.querySelector('.menu-button');
+    menuButton.addEventListener('click', this.toggleMenu.bind(this));
+    window.addEventListener('resize', this.handleResize.bind(this));
+  }
 
-    a {
-      display: flex;
-      gap: 10px;
-      align-items: center;
-      text-decoration: none;
-    }
+  toggleMenu() {
+    const buttonLinks = this.shadowRoot.querySelector('.buttons-container');
+    const menuIcon = this.shadowRoot.querySelector('.menu-button img');
+    const header = this.shadowRoot.querySelector('header');
 
-    a img {
-      width: 45px;
-      height: 22.5px;
+    buttonLinks.classList.toggle('open');
+    header.classList.toggle('vertical');
+    this.classList.toggle('menu-open');
+
+    if (menuIcon.src.includes('menu.png')) {
+      menuIcon.src = 'images/cancel.png';
+    } else {
+      menuIcon.src = 'images/menu.png';
     }
+  }
 
-    a span {
-      font-size: 20px;
-      font-weight: 500;
-      color: black;
+  handleResize() {
+    const buttonLinks = this.shadowRoot.querySelector('.buttons-container');
+    const header = this.shadowRoot.querySelector('header');
+    const menuIcon = this.shadowRoot.querySelector('.menu-button img');
+
+    if (window.innerWidth >= 768) {
+      // Ensure menu is closed in tablet and desktop view
+      buttonLinks.classList.remove('open');
+      header.classList.remove('vertical');
+      this.classList.remove('menu-open');
+      menuIcon.src = 'images/menu.png';
     }
-  </style>
-  <header>
-      <a href="/">
-        <img src="images/logo.png" alt="logo" />
-        <span>달레 스터디</span>
-      </a>
-    
-      <div class="buttons-container">
-        <ds-button-link href="#join-instruction-container" size="small" variant="ghost">참여방법 안내</ds-button-link>
-        <ds-button-link href="https://discord.gg/43UkheRV" size="small" variant="primary">디스코드 참여하기</ds-button-link>
-      </div>
-  </header>
-`;
-
-class HeaderComponent extends HTMLElement {
-  constructor() {
-    super();
+  }
+
+  render() {
     this.attachShadow({ mode: "open" });
-    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.shadowRoot.innerHTML = this.createCss() + this.createHtml();
+  }
+
+  createCss() {
+    return css`
+      :host {
+        position: fixed;
+        top: 0;
+        left: 0;
+        color: var(--text-900);
+        padding: 20px 0;
+        font-weight: 500;
+        width: 100%;
+        border-bottom: 1px solid #d3d3d3;
```

Comment:
여기 border-bottom 도 색상 변경 부탁드려요!

```suggestion
        border-bottom: 1px solid var(--bg-300);
```